iota/api.py

```diff
@@ -1,969 +1,968 @@
 # coding=utf-8
 from __future__ import absolute_import, division, print_function, \
-  unicode_literals
+    unicode_literals
 
-from typing import Dict, Iterable, Optional, Text
```

Comment:
I think we have to add these again as they provide extra functionality for some smart IDEs.

Reply:
Or we leverage `typing` all the way through? But honestly I am not that keen on this. With a proper documentation one should know what he is doing.

https://docs.python.org/3/library/typing.html

Reply:
Type hints are *amazing* 😺

IDEs use them for autocompletion and inspections — they're a great way to boost developer efficiency.

I agree that documentation is important, but they serve different purposes.

Reply:
Hi @todofixthis,
which IDE are you using?
Hope this syntactic excess of ours is not too intrusive. Once this is finished and in `dev`, we tackle the C `pearl_diver`.

Reply:
I use PyCharm, but this is actually a standard practice; see [PEP-484](https://www.python.org/dev/peps/pep-0484/).

The syntax is a bit verbose, because we are supporting Python 2 and Python 3.  If we were only supporting Python 3, we could integrate the type hints directly into the method signature.